java/client/src/org/openqa/selenium/chrome/ChromeOptions.java

```diff
@@ -89,9 +76,78 @@ public ChromeOptions() {
   @Override
   public ChromeOptions merge(Capabilities extraCapabilities) {
     super.merge(extraCapabilities);
+    if (extraCapabilities instanceof ChromeOptions) {
+      Map options = (Map) extraCapabilities.asMap().get("goog:chromeOptions");
```

Comment:
Please use generics. If there's no obvious type, the `Map<?, ?>` prevents compiler warnings, is correct, and is equally meaningless :)